.github/workflows/my-ci-pipeline.yml

```diff
@@ -2,9 +2,30 @@ name: Continuous Integration
 on: [push] 
 jobs: 
  build: 
-  name: Build and test 
+  name: build
   runs-on: ubuntu-latest 
   steps: 
   - uses: actions/checkout@v2 
   - run: docker build --target test --tag todo-app:test .
-  - run: docker run --env-file ./.env.test todo-app:test
\ No newline at end of file
+  - run: docker run --env-file ./.env.test todo-app:test
+ deploy:
+  name: deploy
+  needs: build
+  runs-on: ubuntu-latest
+  if: github.event_name == 'push' && github.ref == 'refs/heads/Module8'
+  steps:
+   - uses: actions/checkout@v2
+   - run: docker login -u keithleverton -p ${{secrets.DOCKERPASSWORD}}
+   - run: docker build --target production --tag keithleverton/todo-app:$GITHUB_SHA .
```

Comment:
Nice work getting this working with the SHA! One hint for the next module though, you'll likely want to push this *and* a latest tag, so that Azure can more easily refresh when you push a new version. You can do that in a single build command by tagging both, and pushing the second image should be quick because of caching, e.g.
```yml
   - run: docker build --target production --tag keithleverton/todo-app:$GITHUB_SHA --tag keithleverton/todo-app .
   - run: docker push keithleverton/todo-app:$GITHUB_SHA
   - run: docker push keithleverton/todo-app
```